lib/filterx/filterx-expr.h

```diff
@@ -68,6 +68,20 @@ struct _FilterXExpr
   gchar *expr_text;
 };
 
+#define FILTERX_EXPR_TYPE_NAME(_type) filterx_expr_type_ ## _type
+
+#define FILTERX_EXPR_DECLARE_TYPE(_type) \
+  extern const gchar *FILTERX_EXPR_TYPE_NAME(_type); \
+  \
+  static inline gboolean \
+  filterx_expr_is_ ## _type(FilterXExpr *expr) \
+  { \
+    return expr && expr->type == FILTERX_EXPR_TYPE_NAME(_type); \
+  }
```

Comment:
It's not easy to find the existence of this function when it's hidden under a macro with such a generic name.

If we really want to use a macro to generate this one-liner function, can we separate it and give it a name related to is_*_type?

I would prefer using explicit functions or just the type comparison expression that is inside this function to avoid more macro-generated code that is difficult to find.

Reply:
I removed the declaration from the macro and defined it for all types.

Reply:
Thank you